tooling/nargo_cli/build.rs

```diff
@@ -63,386 +63,339 @@ const IGNORED_BRILLIG_TESTS: [&str; 11] = [
-        let new_features_ignored = if IGNORED_NEW_FEATURE_TESTS.contains(&test_name.as_str()) {
-            "\n#[ignore]"
-        } else {
-            ""
-        };
-
-        write!(
-            test_file,
-            r#"
-#[test]{new_features_ignored}
-fn execution_success_legacy_{test_name}() {{
-    let test_program_dir = PathBuf::from("{test_dir}");
-
-    let mut cmd = Command::cargo_bin("nargo").unwrap();
-    cmd.arg("--program-dir").arg(test_program_dir);
-    cmd.arg("execute").arg("--force").arg("--use-legacy");
-
-    cmd.assert().success();
-}}
+        }
+        (test_name, dir.path())
+    })
+}
 
+fn generate_test_case(test_file: &mut File, test_type: &str, test_name: &str, test_content: &str) {
+    write!(
+        test_file,
+        r#"
 #[test]
-fn execution_success_{test_name}() {{
-    let test_program_dir = PathBuf::from("{test_dir}");
-
-    let mut cmd = Command::cargo_bin("nargo").unwrap();
-    cmd.arg("--program-dir").arg(test_program_dir);
-    cmd.arg("execute").arg("--force");
-
-    cmd.assert().success();
+fn {test_type}_{test_name}() {{
+    {test_content}
 }}
+"#
+    )
+    .expect("Could not write templated test file.");
+}
 
-#[test]{brillig_ignored}
-fn execution_success_{test_name}_brillig() {{
-    let test_program_dir = PathBuf::from("{test_dir}");
-
-    let mut cmd = Command::cargo_bin("nargo").unwrap();
-    cmd.arg("--program-dir").arg(test_program_dir);
-    cmd.arg("execute").arg("--force").arg("--force-brillig");
+fn generate_execution_success_tests(test_file: &mut File, test_data_dir: &Path) {
+    let test_type = "execution_success";
+    let test_cases = read_test_cases(test_data_dir, test_type);
+    for (test_name, test_dir) in test_cases {
+        let test_dir = test_dir.display();
 
-    cmd.assert().success();
-}}
-            "#,
-            test_dir = test_dir.display(),
-        )
-        .expect("Could not write templated test file.");
+        generate_test_case(
+            test_file,
+            test_type,
+            &test_name,
+            &format!(
+                r#"let test_program_dir = PathBuf::from("{test_dir}");
+
+                let mut cmd = Command::cargo_bin("nargo").unwrap();
+                cmd.arg("--program-dir").arg(test_program_dir);
+                cmd.arg("execute").arg("--force");
+            
+                cmd.assert().success();"#,
+            ),
+        );
+
+        if !IGNORED_NEW_FEATURE_TESTS.contains(&test_name.as_str()) {
+            generate_test_case(
+                test_file,
+                test_type,
+                &format!("legacy_{test_name}"),
+                &format!(
+                    r#"let test_program_dir = PathBuf::from("{test_dir}");
+
+                let mut cmd = Command::cargo_bin("nargo").unwrap();
+                cmd.arg("--program-dir").arg(test_program_dir);
+                cmd.arg("execute").arg("--force").arg("--use-legacy");
```

Comment:
Is there a way to factor out this part as well? This initial setup seems the same in most tests with the slight change of some arguments depending on brillig or the legacy code.

Reply:
I've factored out everything up to the last line.